In [1]:
import pandas as pd

# učitavanje podataka iz CSV datoteke
PATH = "SuperStoreOrders.csv"

# učitavanje svih podataka
data = pd.read_csv(PATH, delimiter=',')

# ispis prvih 5 redova
print(data.head())


          order_id order_date ship_date       ship_mode    customer_name  \
0     AG-2011-2040   1/1/2011  6/1/2011  Standard Class  Toby Braunhardt   
1    IN-2011-47883   1/1/2011  8/1/2011  Standard Class      Joseph Holt   
2     HU-2011-1220   1/1/2011  5/1/2011    Second Class    Annie Thurman   
3  IT-2011-3647632   1/1/2011  5/1/2011    Second Class     Eugene Moren   
4    IN-2011-47883   1/1/2011  8/1/2011  Standard Class      Joseph Holt   

       segment            state    country  market   region  ...  \
0     Consumer      Constantine    Algeria  Africa   Africa  ...   
1     Consumer  New South Wales  Australia    APAC  Oceania  ...   
2     Consumer         Budapest    Hungary    EMEA     EMEA  ...   
3  Home Office        Stockholm     Sweden      EU    North  ...   
4     Consumer  New South Wales  Australia    APAC  Oceania  ...   

          category sub_category                 product_name sales quantity  \
0  Office Supplies      Storage          Tenex Lockers,

In [2]:
# Ispis dimenzija skupa podataka (potrebno je učitati sve podatke radi analize, ne samo prvih 2000 redova)
print("\nDimenzije skupa podataka (broj redaka i stupaca):")
print(data.shape)


Dimenzije skupa podataka (broj redaka i stupaca):
(51290, 21)


In [3]:
# Ispis imena stupaca i nedostajućih vrijednosti
print(data.isna().sum())

order_id          0
order_date        0
ship_date         0
ship_mode         0
customer_name     0
segment           0
state             0
country           0
market            0
region            0
product_id        0
category          0
sub_category      0
product_name      0
sales             0
quantity          0
discount          0
profit            0
shipping_cost     0
order_priority    0
year              0
dtype: int64


In [4]:
# Ispis broja jedinstvenih vrijednosti po stupcima
print(data.nunique())

order_id          25035
order_date         1430
ship_date          1464
ship_mode             4
customer_name       795
segment               3
state              1094
country             147
market                7
region               13
product_id        10292
category              3
sub_category         17
product_name       3788
sales              2246
quantity             14
discount             27
profit            24575
shipping_cost     10037
order_priority        4
year                  4
dtype: int64


In [5]:
# Ispis tipova podataka po stupcima (analiza kvanitativnih i kvalitativnih varijabli)
print(data.dtypes)

order_id           object
order_date         object
ship_date          object
ship_mode          object
customer_name      object
segment            object
state              object
country            object
market             object
region             object
product_id         object
category           object
sub_category       object
product_name       object
sales              object
quantity            int64
discount          float64
profit            float64
shipping_cost     float64
order_priority     object
year                int64
dtype: object


In [ ]:
# Ispis broja jedinstvenih vrijednosti po stupcima
for column in data:
    print(data[column].value_counts())
    input("...")

order_id
CA-2014-100111    14
MX-2014-166541    13
TO-2014-9950      13
NI-2014-8880      13
IN-2013-42311     13
                  ..
PL-2013-7560       1
US-2013-144050     1
CA-2013-138933     1
US-2013-165022     1
CA-2014-156720     1
Name: count, Length: 25035, dtype: int64
order_date
18-06-2014    135
18-11-2014    127
3/9/2014      126
20-11-2014    118
29-12-2014    116
             ... 
13-01-2013      1
10/11/2013      1
2/1/2011        1
27-05-2012      1
23-03-2014      1
Name: count, Length: 1430, dtype: int64
ship_date
22-11-2014    130
7/9/2014      115
17-11-2014    101
7/12/2014     101
29-11-2014    100
             ... 
14-04-2011      3
25-01-2011      3
3/1/2011        2
5/1/2011        2
6/1/2011        2
Name: count, Length: 1464, dtype: int64
ship_mode
Standard Class    30775
Second Class      10309
First Class        7505
Same Day           2701
Name: count, dtype: int64
customer_name
Muhammed Yedwab        108
Steven Ward            106
Patrick O'Brill       

In [11]:
# Ispis imena stupaca
print(data.columns.values)

['Diabetes_012' 'HighBP' 'HighChol' 'CholCheck' 'BMI' 'Smoker' 'Stroke'
 'HeartDiseaseorAttack' 'PhysActivity' 'Fruits' 'Veggies'
 'HvyAlcoholConsump' 'AnyHealthcare' 'NoDocbcCost' 'GenHlth' 'MentHlth'
 'PhysHlth' 'DiffWalk' 'Sex' 'Age' 'Education' 'Income']


# Pitanja:

### 1) Je li je skup podataka dovoljno velik? (Optimalno >15 000 redaka i >20 stupaca)

Da. Skup podataka ima 51,290 redaka i 21 stupac.

### 2) Je li skup ima dovoljno različite podatke?

Da, skup ima dovoljno različite podatke.

### 3) Je li skup ima vremensku dimenziju?

Da.

Primjer:

order_date (1430 jedinstvenih vrijednosti) - datum narudžbe

ship_date (1464 jedinstvenih vrijednosti) - datum isporuke

year (4 godine: 2011, 2012, 2013, 2014) - godina narudžbe

### 4) Je li skup ima kvantitativne i kvalitativne podatke?

Da. 

*Kvantitativne varijable*:

quantity (int64) - količina naručenih proizvoda

discount (float64) - popust na proizvod

profit (float64) - ostvareni profit

shipping_cost (float64) - trošak dostave

sales (iako je kao object, predstavlja numeričke vrijednosti prodaje)


<br>

*Kvalitativne varijable:*



kategorijske: category, sub_category, segment, ship_mode, order_priority, market, region

identifikatori: order_id, product_id, customer_name

geografske: country, state

vremenske: order_date, ship_date (kao objekti, ali predstavljaju datume)

### 5) Je li skup ima puno nedostajućih vrijednosti?

 Ne, skup podataka nema nijednu nedostajuću vrijednost.

